graf2d/gpad/src/TPad.cxx

```diff
@@ -584,7 +584,7 @@ TLegend *TPad::BuildLegend(Double_t x1, Double_t y1, Double_t x2, Double_t y2,
             leg = new TLegend(x1, y1, x2, y2, title);
          TList * hlist = ((THStack *)o)->GetHists();
          TIter nexthist(hlist);
-         while (auto obj = nexthist()) {
+         while ((obj = nexthist())) {
```

Comment:
For clarity of the code and the scope of variable I would have keep this line as is, remove the declaration on 509 and update line 566 and 516 to use their own local declaration.